specification/dartLangSpec.tex

```diff
@@ -3073,20 +3076,32 @@ \subsection{Constructors}
 where $X_j, j \in 1 .. m$ are the formal type parameters of $C$
 and $T_j, j \in 1 .. m$ are specified in the given context.
 We may also omit such a substitution when the given context is
-the instance scope of $C$, where $X_1, \ldots, X_m$ are in scope.
+the class scope of $C$, where $X_1, \ldots, X_m$ are in scope.
 }
 
-\commentary{
+\commentary{%
 A constructor declaration may conflict with static member declarations
 (\ref{classMemberConflicts}).
 }
 
-% In what scope do constructors go? The simple names of named constructors go in the static scope of the class. Unnamed ones go nowhere, but we use the class name to refer to them; the class name could also in the static scope of the class as well to prevent weird errors, or we could ban it explicitly and avoiding duplication. Similarly, the instance scope could contain the constructor names and class name, or we could have special rules to prevent collisions between instance members and constructors or the class.
-
-% The enclosing scope of a generative constructor is the instance scope of the class in which it is declared (but what about redirecting?)
+\commentary{%
+A constructor declaration does not introduce a name into a scope.
+The resolution of an instance creation expression
+(\ref{new})
+to a particular constructor
+relies on the library scope to determine the class
+(possibly via an import prefix),
+and for a constructor denoted by \code{$C$.\id} or \code{$prefix$.$C$.\id},
+it is directly specified that the class $C$ must declare
+a constructor with the name \code{$C$.\id}.
+In other words, the constructor is never determined via a lexical lookup,
+only the class.%
+}
 
 \LMHash{}%
-If{}f no constructor is specified for a class $C$, it implicitly has a default constructor \code{C() : \SUPER{}() \{\}}, unless $C$ is class \code{Object}.
+If{}f no constructor is specified for a class $C$,
+it implicitly has a default constructor \code{C():\ \SUPER()\ \{\}},
+unless $C$ is the built-in class \code{Object}.
```

Comment:
> ... is one of the built-in classes \code{Object} or \code{Null}.

(If we are post-null-safety).

Reply:
Thanks! I added a TODO comment to do that.